### PR TITLE
profile page update

### PR DESCRIPTION
**Pull request by jbterrylin**

### CODE REVIEW

File: app/components/Pagination/index.js [hunk 1]
```diff
@@ -12,55 +12,63 @@ class Pagination extends React.PureComponent { // eslint-disable-line react/pref
     }
 
     componentWillMount() {
-        if (this.props.goToPage && this.props.goToPage !== 1) {
-            let selfApiUrl = this.props.parentProps.herlisting.data.product.result._links.self.href;
-            if (selfApiUrl.indexOf('page=') !== -1) {
-                selfApiUrl = selfApiUrl.replace(`page=${this.props.parentProps.herlisting.data.product.result._meta.currentPage}`, `page=${this.props.goToPage}`);
-            } else {
-                selfApiUrl += `${selfApiUrl.indexOf('?' !== -1) ? '&' : '?'}page=${this.props.goToPage}`;
+        if (!this.props.checking) {
```
Comment (tienping) [inline app/components/Pagination/index.js]: change to this.props.isHerlisting, boolean

File: app/components/ProductCard/index.js [hunk 2]
```diff
@@ -4,93 +4,127 @@
...
-                            </div>
+            <Price
+                currency={this.props.product.currency}
+                price={this.props.price}
+            />
+        );
+    }
+
+    renderCross = () => {
+        if (!this.props.cross) {
```
Comment (tienping) [inline app/components/ProductCard/index.js]: this.props.allowDelete, boolean

File: app/components/ProductCard/index.js [hunk 3]
```diff
@@ -4,93 +4,127 @@
...
 
+    render() {
+        return (
+            <div
+                className={`product-card ${this.props.listViewMode ? 'grid-view-card' : 'list-view-card'}`}
+                key={this.props.product.id}
+            >
+                <div className={`product-card-content ${!this.props.product.instock ? '' : 'OOS'}`}>
+                    <div className="product-card-images">
+                        {this.renderCross()}
```
Comment (tienping) [inline app/components/ProductCard/index.js]:
```
{
    this.props.allowDelete &&
        <div onClick={() => this.props.deleteFromWishlist()}>
            <i className="fa fa-times cross-icon" aria-hidden="true"></i>
        </div>
}
```

File: app/components/ProductCard/index.js [hunk 4]
```diff
@@ -4,93 +4,127 @@
...
-                        </div>
-                        <div className="product-card-info">
+                        </NavLink>
+                    </div>
+                    <div className="product-card-info">
+                        <NavLink to={this.props.url}>
                             <div className="product-card-wishlist">
                                 <i
-                                    className={`wishlist-btn fa fa-heart ${this.state.wishlist ? 'wishlist-btn-clicked' : ''}`}
+                                    className={`wishlist-btn fa fa-heart ${true ? 'wishlist-btn-clicked' : ''}`}
```
Comment (tienping) [inline app/components/ProductCard/index.js]: fix this hardcoded value

File: app/containers/App/index.js [hunk 5]
```diff
@@ -105,7 +132,14 @@ export class App extends React.PureComponent { // eslint-disable-line react/pref
...
                         />
                         <Route exact={true} path="/mall/:productId" component={ProductView} />
+                        <Route exact={true} path="/profile/me" component={ProfileEditInform} />
+                        <Route exact={true} path="/order" component={ProfileOrder} />
+                        <Route exact={true} path="/wishlist" component={ProfileWishlist} />
+                        <Route exact={true} path="/wallet" component={ProfileWallet} />
+                        <Route exact={true} path="/review" component={ProfileReview} />
+                        <Route exact={true} path="/about" component={AboutUs} />
                         <Route exact={true} path="/" component={HomePage} />
+                        <Route exact={true} path="/profile1/" component={ProfileWholePage} />
```
Comment (tienping) [inline app/containers/App/index.js]: profile1??

File: app/containers/ProfileReview/index.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        category: '/mall/reviewable?reviewed=0&per-page=5',
+    }
+
+    componentWillMount() {
+        this.props.dispatch(getReview('/mall/reviewable?reviewed=0&per-page=5', 'reviewData'));
+    }
+
+    insertRateDataToReviewData = (mark, id) => {
+        const obj = { ...this.state.reviewData };
+        obj[`${id}`] = mark;
```
Comment (tienping) [inline app/containers/ProfileReview/index.js]: obj[id] = mark;

**Review by tienping — CHANGES_REQUESTED** [review]

**Review by tienping — APPROVED** [review]

**Comment by jbterrylin** [comment]
fix code done

**Pull request merged**